nemo/collections/asr/models/ssl_models.py

```diff
@@ -460,6 +457,11 @@ def decoder_loss_step(self, spectrograms, spec_masks, encoded, encoded_len, targ
 
     # PTL-specific methods
     def training_step(self, batch, batch_nb):
+        if self.output_from_layer is not None and len(self.output_from_layer) > 0:
```

Comment:
output_from_layer is a dict of loss name -> None or string, so we should check here that at least one of the value is not None; also need to check that self.decoder_losses is not None, as otherwise we only have one loss so output_from_layer is not defined